e2e/vegeta_test.go

```diff
@@ -0,0 +1,824 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"os"
+	"slices"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/cheggaaa/pb/v3"
+	vegeta "github.com/tsenart/vegeta/lib"
+
+	"github.com/buildwithgrove/path/protocol"
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+	"github.com/buildwithgrove/path/request"
+)
+
+/* -------------------- ANSI Color Constants -------------------- */
+
+// ANSI color codes for terminal output
+const (
+	RED       = "\x1b[31m"
+	GREEN     = "\x1b[32m"
+	YELLOW    = "\x1b[33m"
+	BLUE      = "\x1b[34m"
+	BOLD      = "\x1b[1m"
+	BOLD_BLUE = "\x1b[1m\x1b[34m"
+	BOLD_CYAN = "\x1b[1m\x1b[36m"
+	RESET     = "\x1b[0m"
+)
+
+/* -------------------- Vegeta Helper Functions -------------------- */
+
+// createRPCTarget creates a vegeta.Targeter for the specified RPC method
+func createRPCTarget(
+	gatewayURL string,
+	serviceID protocol.ServiceID,
+	jsonrpcReq jsonrpc.Request,
+) vegeta.Targeter {
+	return func(tgt *vegeta.Target) error {
+		body, err := json.Marshal(jsonrpcReq)
+		if err != nil {
+			return err
+		}
+
+		tgt.Method = http.MethodPost
+		tgt.URL = gatewayURL
+		tgt.Body = body
+		tgt.Header = http.Header{
+			"Content-Type":                    []string{"application/json"},
+			request.HTTPHeaderTargetServiceID: []string{string(serviceID)},
+		}
+
+		return nil
+	}
+}
+
+// runAttack executes a load test for the given method
+func runAttack(
```

Comment:
I recommend a more descriptive name here.

Reply:
The term"attack" comes from the Vegeta library, which specifically calls sending a request an attack. WDYT about maybe just adding a comment that adds this context?